src/test/java/PasswordGeneratorTest.java

```diff
@@ -1,18 +1,94 @@
 /*
  * This Java source file was generated by the Gradle 'init' task.
  */
+
+import org.assertj.core.api.AbstractBooleanAssert;
+import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Test;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class PasswordGeneratorTest {
-    @Test public void checkSecretPassword() {
-        String secretPassword = "";
 
+    private PasswordGenerator p;
+
+    @Before
+    public void setUp() {
+        p = new PasswordGenerator();
+    }
+
+    @Test
+    public void shouldReturnSecretPassword() {
+        String secretPassword = p.checkSecretPassword(3, 1, 1);
         BCryptPasswordEncoder bCryptPasswordEncoder = new BCryptPasswordEncoder();
+
         assertThat(bCryptPasswordEncoder.matches(secretPassword,
                 "$2a$10$N43LdqU9b1ZMuZM2KiBnIeaXiPqxYcTEVoMFabb9ZV2jfG82jupD6"))
                 .isTrue();
     }
+
+    @Test
+    public void shouldReturnRandomLetter() {
+        assertThat(p.RandomIndexGenerator(p.getAlphaCharArray())).isNotEqualTo(p.RandomIndexGenerator(p.getAlphaCharArray()));
```

Comment:
Not clear what this test is trying to assert, it looks like it's checking that two calls to generate a random letter return a different result?  If so it's a bad test, there's nothing to stop the same random character being generated by consecutive calls, hence it won't reliably pass.